src/main/kotlin/com/software/modsen/driverservice/service/DriverService.kt

```diff
@@ -0,0 +1,91 @@
+package com.software.modsen.driverservice.service
+
+import com.software.modsen.driverservice.dto.request.DriverRequest
+import com.software.modsen.driverservice.dto.request.toDriver
+import com.software.modsen.driverservice.dto.response.DriverListResponse
+import com.software.modsen.driverservice.dto.response.DriverResponse
+import com.software.modsen.driverservice.dto.response.DriverWithCarResponse
+import com.software.modsen.driverservice.exception.CarNotFoundException
+import com.software.modsen.driverservice.exception.DriverNotFoundException
+import com.software.modsen.driverservice.exception.EmailAlreadyExistException
+import com.software.modsen.driverservice.exception.PhoneAlreadyExistException
+import com.software.modsen.driverservice.model.Car
+import com.software.modsen.driverservice.model.Driver
+import com.software.modsen.driverservice.model.toDriverResponse
+import com.software.modsen.driverservice.model.toDriverWithCarResponse
+import com.software.modsen.driverservice.repository.CarRepository
+import com.software.modsen.driverservice.repository.DriverRepository
+import com.software.modsen.driverservice.util.ExceptionMessages
+import lombok.RequiredArgsConstructor
+import org.springframework.stereotype.Service
+
+@Service
+@RequiredArgsConstructor
+class DriverService(
+    private val driverRepository: DriverRepository,
+    private val carRepository: CarRepository
+) {
+    fun getDriverById(id: Long): DriverWithCarResponse {
+        return getByIdOrElseThrow(id).toDriverWithCarResponse()
+    }
+
+    fun getAllDrivers(): DriverListResponse {
+        return DriverListResponse(driverRepository.findAll()
+            .map { it.toDriverResponse() })
+    }
+
+    fun createDriver(driverRequest: DriverRequest): DriverResponse {
+        preCreateValidateDriver(driverRequest)
+        val driver: Driver = driverRequest.toDriver()
+        driver.car = getCarByIdOrElseThrow(driverRequest.carId)
+        return driverRepository.save(driver).toDriverResponse()
+    }
+
+    fun updateDriver(id: Long, driverRequest: DriverRequest): DriverResponse {
+        val driver_opt: Driver = getByIdOrElseThrow(id)
```

Comment:
Do not use snake case in kotlin or java apps

Reply:
Fixed